src/jobs/deploy_to_selfhosted.yml

```
@@ -0,0 +1,38 @@
+description: "A job to deploy latest dev components to self-hosted"
+
+executor: do
+
+steps:
+  - deploy:
+      name: Trigger workflow to deploy dev components
+      command: |
+        response=$(curl -XPOST -s -H 'Accept: application/json' -H 'x-attribution-login: string' -H 'x-attribution-actor-id: string' -H 'Content-type: application/json' -d '{"branch": "master"}' 'https://circleci.com/api/v2/project/gh/codacy/chart/pipeline' -H "Circle-Token: $CIRCLE_CI_GITHUB_TOKEN")
+        workflowState=$(echo $response | jq '.state' | sed 's/\"//g')
+        workflowID=$(echo $response | jq '.id' | sed 's/\"//g')
+
+        if [ "$workflowState" != "pending" ] && [ "$workflowState" != "running" ] ; then
+            echo "Remote workflow execution failed abruptly!"
+            echo $response
+            exit 1
+        fi
+
+        echo "Polling for workflow id: $workflowID"
+        getWorkflowCommand="curl -s -H 'Accept: application/json' -H 'Content-type: application/json' \"https://circleci.com/api/v2/pipeline/$workflowID/workflow\" -H \"Circle-Token: $CIRCLE_CI_GITHUB_TOKEN\""
+        status=$(eval $getWorkflowCommand | jq '.items[0].status' | sed "s/\"//g")
+        echo $status
+        while [[ "$status" == "pending" || "$status" == "running" || "$status" == "null" ]]
+        do
+            echo $status
```

I would add a message before the status, just so that it easier to understand when seeing the logs.

```suggestion
            echo "Workflow status: $status"
```

I would also do the same above